pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -561,40 +561,65 @@ func collectDiskMetricsFromPVC(vm *k6tv1.VirtualMachine) []operatormetrics.Colle
 	var cr []operatormetrics.CollectorResult
 
 	for _, vol := range vm.Spec.Template.Spec.Volumes {
-		if vol.PersistentVolumeClaim != nil {
-			diskName := vol.Name
-			pvcName := vol.PersistentVolumeClaim.ClaimName
-
+		if pvcName, diskName, isDataVolume := getPVCAndDiskName(vol); pvcName != "" {
 			key := controller.NamespacedKey(vm.Namespace, pvcName)
-			obj, exists, err := persistentVolumeClaimInformer.GetStore().GetByKey(key)
-			if err != nil {
+
+			if obj, exists, err := persistentVolumeClaimInformer.GetStore().GetByKey(key); err != nil {
 				log.Log.Errorf("Error retrieving PVC %s in namespace %s: %v", pvcName, vm.Namespace, err)
-				continue
-			}
-			if !exists {
+			} else if !exists {
 				log.Log.Warningf("PVC %s in namespace %s does not exist", pvcName, vm.Namespace)
-				continue
-			}
-			pvc, ok := obj.(*k8sv1.PersistentVolumeClaim)
-			if !ok {
+			} else if pvc, ok := obj.(*k8sv1.PersistentVolumeClaim); !ok {
 				log.Log.Warningf("Object for PVC %s in namespace %s is not of expected type", pvcName, vm.Namespace)
-				continue
+			} else {
+				cr = append(cr, GetDiskSizeValues(vm, pvc, diskName, isDataVolume))
 			}
+		}
+	}
+	return cr
+}
+
+func getPVCAndDiskName(vol k6tv1.Volume) (pvcName, diskName string, isDataVolume bool) {
+	if vol.PersistentVolumeClaim != nil {
+		return vol.PersistentVolumeClaim.ClaimName, vol.Name, false
+	}
+	if vol.DataVolume != nil {
+		return vol.DataVolume.Name, vol.Name, true
+	}
+	return "", "", false
+}
+
+func GetDiskSizeValues(vm *k6tv1.VirtualMachine, pvc *k8sv1.PersistentVolumeClaim, diskName string, isDataVolume bool) operatormetrics.CollectorResult {
+	var pvcSize *resource.Quantity
 
-			pvcSize := pvc.Spec.Resources.Requests.Storage()
-			volumeMode := "null"
-			if pvc.Spec.VolumeMode != nil {
-				volumeMode = string(*pvc.Spec.VolumeMode)
+	if isDataVolume {
+		pvcSize = getSizeFromDataVolumeTemplates(vm, pvc.Name)
+	}
+	if pvcSize == nil {
+		pvcSize = pvc.Spec.Resources.Requests.Storage()
+	}
+
+	volumeMode := "null"
```

Comment:
as we use for other labels, maybe

```suggestion
	volumeMode := "<none>"
```